content/help/ops/security/debugging-authorization/index.md

```diff
@@ -0,0 +1,237 @@
+
+Pilot is responsible for converting and distributing your authorization policies to proxies. Follow
+the below steps to make sure this is finished as expected:
+
+1. Turn on debug logging for authorization in Pilot
+
+    First export the Pilot `ControlZ` page with the following command:
+
+    {{< text bash >}}
+    $ kubectl port-forward $(kubectl -n istio-system get pods -l istio=pilot -o jsonpath='{.items[0].metadata.name}') -n istio-system 9876:9876
+    Forwarding from 127.0.0.1:9876 -> 9876
+    {{< /text >}}
+
+    Then start your browser and open the `ControlZ` page at `http://127.0.0.1:9876/scopez/`. Change the
+    `rbac` Output Level to `debug`. After this, use `Ctrl+C` to stop the port-forward command.
+
+1. Check the related authorization debug logging in Pilot
+
+    > Note: You probably need to first delete and then re-apply your authorization policies so that
+the debug output is generated for these policies.
+
+    Check Pilot log with the following command:
+
+    {{< text bash >}}
+    $ kubectl logs $(kubectl -n istio-system get pods -l istio=pilot -o jsonpath='{.items[0].metadata.name}') -c discovery -n istio-system | grep rbac
+    ... ...
+    2018-07-26T22:25:41.009838Z debug rbac building filter config for {sleep.foo.svc.cluster.local map[app:sleep pod-template-hash:3326367878] map[destination.name:sleep destination.namespace:foo destination.user:default]}
+    2018-07-26T22:25:41.009915Z info  rbac no service role in namespace foo
+    2018-07-26T22:25:41.009957Z info  rbac no service role binding in namespace foo
+    2018-07-26T22:25:41.010000Z debug rbac generated filter config: { }
+    2018-07-26T22:25:41.010114Z info  rbac built filter config for sleep.foo.svc.cluster.local
+    2018-07-26T22:25:41.182400Z debug rbac building filter config for {productpage.default.svc.cluster.local map[pod-template-hash:2600844901 version:v1 app:productpage] map[destination.name:productpage destination.namespace:default destination.user:bookinfo-productpage]}
+    2018-07-26T22:25:41.183131Z debug rbac checking role app2-grpc-viewer
+    2018-07-26T22:25:41.183214Z debug rbac role skipped for no AccessRule matched
+    2018-07-26T22:25:41.183255Z debug rbac checking role productpage-viewer
+    2018-07-26T22:25:41.183281Z debug rbac matched AccessRule[0]
+    2018-07-26T22:25:41.183390Z debug rbac generated filter config: {policies:<key:"productpage-viewer" value:<permissions:<and_rules:<rules:<or_rules:<rules:<header:<name:":method" exact_match:"GET" > > > > > > principals:<and_ids:<ids:<any:true > > > > >  }
+    2018-07-26T22:25:41.184407Z info  rbac built filter config for productpage.default.svc.cluster.local
+    ... ...
+    {{< /text >}}
+
+    To make sure Pilot is handling the authorization policies correctly, look at the output to:
+
+    * Carefully check if there are any errors in the log.
+    * Check if there is a `built filter config for` message which means a filter config is generated
```

Comment:
Make numbered steps.
1. Check if there are any errors in the log.
1. Check if there is a `built filter config for` message which means a filter config is generated for the target service.